docs/source/developers/guide/step_by_step/pr_and_github.rst

```diff
@@ -24,14 +24,162 @@
 
 .. _pr_and_github:
 
-****************************
-Lifecycle of a Pull Request
-****************************
+******************************
+Lifecycle of a Pull Request 🙀 
+******************************
 
+:ref:`As mentioned before<set-up>`, Arrow project uses git for
+version control and a workflow based on Pull Requests. That means
+that you contribute the changes to the code by creating a branch
+in Git, make changes to the code, push the changes to you ``origin``
+which is your fork of the Arrow repository on GitHub and then you
+create a **Pull Request** against the official Arrow repository
+which is saved in your set up as ``upstream``.
 
-Creating a PR 🙀
-================
+You should have git set up by now, have your source code,
+have successfully built Arrow and have an JIRA issue to work on.
 
+**Before making changes to the code, you should create a new
+branch in Git.**
 
-Reviews and get the PR merge 🎉
-===============================
\ No newline at end of file
+1. Update/sync the code from your ``upstream``
+   in the master branch. Run it in the shell from ``arrow`` directory.
+
+   .. code-block:: shell
+
+      $ git checkout master
+      $ git fetch upstream
+      $ git pull --ff-only upstream master
+
+2. Create a new branch
+
+   .. code-block:: shell
+
+      $ git checkout -b <branch-name>
+
+Now you can make changes to the code. To see the changes
+made in the library use this two commands:
+
+.. code-block:: shell
+
+   $ git status # to see what files are changed
+   $ git diff # to see code change per file
+
+Creating a Pull Request 
+=======================
+
+Once you are satisfied with the changes, run the :ref:`tests <testing>`
+and linters and then go ahead and commit the changes.
+
+3. Add and commit the changes
```

Comment:
Just wondering, do those numbers work? (it might be that sphinx automatically renumbers them and starts counting at 1 because this is a new list. But if that's the case, there should also be a way to turn that off)

Reply:
It works. If I would use `#.` it would restart at 1.